dot/core/service.go

```diff
@@ -372,17 +380,33 @@ func (s *Service) maintainTransactionPool(block *types.Block) {
 		s.transactionState.RemoveExtrinsic(ext)
 	}
 
+	bestBlockHash := s.blockState.BestBlockHash()
+	stateRoot, err := s.storageState.GetStateRootFromBlock(&bestBlockHash)
+	if err != nil {
+		logger.Errorf("could not get state root from block %s: %w", bestBlockHash, err)
+	}
+
+	ts, err := s.storageState.TrieState(stateRoot)
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+
 	// re-validate transactions in the pool and move them to the queue
 	txs := s.transactionState.PendingInPool()
 	for _, tx := range txs {
-		// get the best block corresponding runtime
-		rt, err := s.blockState.GetRuntime(nil)
+		rt, err := s.blockState.GetRuntime(&bestBlockHash)
 		if err != nil {
 			logger.Warnf("failed to get runtime to re-validate transactions in pool: %s", err)
 			continue
 		}
 
-		txnValidity, err := rt.ValidateTransaction(tx.Extrinsic)
+		rt.SetContextStorage(ts)
+		externalExt, err := s.buildExternalTransaction(rt, tx.Extrinsic)
+		if err != nil {
+			logger.Errorf("Unable to build external transaction: %s", err)
```

Comment:
Shouldn't we return an error instead here?
Do  we really want to continue if one extrinsic fails to build/validate?

Reply:
I mean I think we do, see my explanation for why in an above comment. Let me know if my logic seems flawed though

Reply:
Shouldn't we `continue` to the next transaction? Why are we passing it to the runtime to validate if we can't build it?